.github/workflows/ci.yml

```diff
@@ -0,0 +1,66 @@
+name: CI Pipeline
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+env:
+  GOLANGCI_LINT_VERSION: v1.61.0
+
+jobs:
```

Comment:
Is there a problem using one job for lint, build and test? Now it creates one check per job on the PR, would simplify to only have one job with multiple steps when configuring branch protection rules, e.g. which checks to be required
<img width="758" alt="image" src="https://github.com/user-attachments/assets/97bc7b65-37f0-49d4-b571-c2355e0f61fe">

Reply:
It was recommended by the golangci-lint docs to use separate job - https://github.com/golangci/golangci-lint-action?tab=readme-ov-file#how-to-use

Reply:
Though it works in the single job.

Reply:
Hmm okay 🤔 Guess it depends on if you want to run lint in parallel with build/test vs all three synchronously. No strong opinions really, it was more convenient for having one required check and configuring the branch protection

Reply:
Merged into single job.

Reply:
Ready to rumble. :heavy_check_mark: